src/sentry/api/endpoints/sentry_app_publish_request.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry import options
+from sentry import features
+from sentry.api.bases.sentryapps import SentryAppBaseEndpoint
+from sentry.utils.email import send_mail
+from sentry.utils import json
+
+
+class SentryAppPublishRequestEndpoint(SentryAppBaseEndpoint):
+    def put(self, request, sentry_app):
+        if not features.has('organizations:sentry-apps',
+                            sentry_app.owner,
+                            actor=request.user):
+
+            return Response(status=404)
+
+        message = 'User %s of organization %s wants to publish %s' % (
+            request.user, sentry_app.owner.slug, sentry_app.slug)
+
+        print(message)
+
+        send_mail(
+            'Sentry App Publication Request',
+            message,
+            options.get('mail.from'),
+            ['partners@sentry.io'],
+            fail_silently=False
+        )
```

Comment:
I'm not sure how we typically use `send_mail` but do we want to be able to have the email come from the user's email so that we can respond to them? If so, would we want them to be able to input their email of choice?

Or do we just want this to be internal for us to know who has requested, and we can reach out to them once we receive the email ?

Reply:
@MeredithAnya I think that modifying the from address to an email we don't control isn't a great idea. However, we can add a `reply-to` header and put the user's email there. I'd need to modify the core `send_email` function to do that. @mnoble Thoughts?

Reply:
I think either one works, tbh. I wouldn't worry about allowing them to choose which email. We can just use the one on their `User` and trust they check it. As far as who this email comes from, I don't think it matter. It's fine just being internal since the message and all is pretty raw and not super user-friendly (if we replied, they'd see it).